### PR TITLE
Support for Heading sensor

### PR DESCRIPTION
**Pull request by Adi-MMandroid**
### Tested on:
Device :-

### CODE REVIEW

**Comment by Adi-MMandroid** [comment]
@shubhamsinghmutualmobile not able to get this sensor to work on my device, isAvailable is false, could you get someone to test this please?

File: composesensors/src/main/java/com/mutualmobile/composesensors/HeadingSensorState.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * Measures a device's heading value in degrees (anti-clockwise),
+ * @param degrees Indicates direction in which the device is pointing relative to true north in degrees. Defaults to 0f.
+ * @param confidence Indicates the confidence of prediction, under Gaussian standard normal distribution. Defaults to 0f.
+ * @param isAvailable Whether the current device has a heading sensor. Defaults to false.
+ */
+@Immutable
+class HeadingSensorState(
+    val degrees: Float = 0f,
+    val confidence: Float = 0f,
+    val isAvailable: Boolean = false
```
Comment (shubhamsinghmutualmobile) [inline composesensors/src/main/java/com/mutualmobile/composesensors/HeadingSensorState.kt:19]: We also include the `accuracy` parameter in all our state classes in ComposeSensors. Refer to any one of them for more info.

File: composesensors/src/main/java/com/mutualmobile/composesensors/HeadingSensorState.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import androidx.compose.runtime.remember
+
+/**
+ * Measures a device's heading value in degrees (anti-clockwise),
+ * @param degrees Indicates direction in which the device is pointing relative to true north in degrees. Defaults to 0f.
+ * @param confidence Indicates the confidence of prediction, under Gaussian standard normal distribution. Defaults to 0f.
+ * @param isAvailable Whether the current device has a heading sensor. Defaults to false.
+ */
+@Immutable
+class HeadingSensorState(
```
Comment (shubhamsinghmutualmobile) [inline composesensors/src/main/java/com/mutualmobile/composesensors/HeadingSensorState.kt:16]: Also, we want this class' constructor to be internal as we want our users to make use of the composable function only.

**Comment by shubhamsinghmutualmobile** [comment]
@Adi-MMandroid We can remove the `[WIP]` prefix from this PR's title 👍

**Review by shubhamsinghmutualmobile — CHANGES_REQUESTED** [review]
@Adi-MMandroid Let's address the comments and also make the required change to the `README.md` file and mark this sensor's state as ✅ there.

**Comment by shubhamsinghmutualmobile** [comment]
Commit history will be cleaner if we squash these commits ✅

**Pull request closed without merging**